dolphin/drivers/dell_emc/vmax/client.py

```diff
@@ -123,3 +123,58 @@ def list_pools(self):
             LOG.error("Failed to get pool metrics from VMAX: {}".format(err))
             raise exception.StorageBackendException(
                 reason='Failed to get pool metrics from VMAX')
+
+    def list_volumes(self, storage_id):
+
+        try:
+            # List all volumes
+            volumes = self.conn.provisioning.get_volume_list(filters={'data_volume': 'false'})
+
+            volume_list = []
+            for volume in volumes:
+                # Get volume details
+                vol = self.conn.provisioning.get_volume(volume)
+
+                total_cap = vol['cap_mb'] * units.Mi
+                used_cap = (total_cap * vol['allocated_percent']) / 100.0
+                free_cap = total_cap - used_cap
+
+                # TODO: Update constants.VolumeStatus to make mapping more precise
+                switcher = {
+                    'Ready': constants.VolumeStatus.AVAILABLE,
+                    'Not Ready': constants.VolumeStatus.ERROR,
+                    'Mixed': constants.VolumeStatus.ERROR,
+                    'Write Disabled': constants.VolumeStatus.ERROR,
+                    'N/A': constants.VolumeStatus.ERROR,
+                }
+                status = switcher.get(vol['status'], constants.VolumeStatus.ERROR)
+
+                v = {
+                    "name": volume,
+                    "storage_id": storage_id,
+                    "description": vol['type'],
```

Comment:
Do we use 'description' field for storing Type of volumes?

Reply:
We have not defined what is required in description field. So just passing type as additional info.

Reply:
This is misleading

Reply:
I hope we can keep 'type' in description, as we don't have a field to represent type .. or we can map the type to comprehensive description like 
R11-> A concurrent R1 device with 2 R1 SRDF mirrors.

Reply:
This is fine. But ‘description’ and ‘type’ has different meaning. Description can be ‘This volume is used for ...’ but type is the volume type which is an attribute. Overloading may mislead. But why dont we have separate Type field

Reply:
Updated description